mono/metadata/w32error-unix.c

```diff
@@ -54,7 +54,9 @@ mono_w32error_unix_to_win32 (guint32 error)
 	case ENOENT:
 	case ENOTDIR: return ERROR_FILE_NOT_FOUND;
 	case ENOSPC: return ERROR_HANDLE_DISK_FULL;
+#if !defined(_AIX) || (defined(_AIX) && defined(_LINUX_SOURCE_COMPAT))
 	case ENOTEMPTY: return ERROR_DIR_NOT_EMPTY;
```

Comment:
Can we ifdef ENOTEMPTY? It is never const int, always define, right?

Reply:
There are advantages/disadvantages either way -- with ifdef, you use it whenever AIX adds it, without ifdef you err toward trying to use it and noticing when it is absent.

Reply:
AIX can return this, but only with a certain ifdef enabled, as this shows. (don't use it because of possible semantics changes, but could consider it)